app/controllers/api/messaging.php

```diff
@@ -313,24 +313,23 @@
     ->label('sdk.response.model', Response::MODEL_PROVIDER)
     ->param('providerId', '', new CustomId(), 'Provider ID. Choose a custom ID or generate a random ID with `ID.unique()`. Valid chars are a-z, A-Z, 0-9, period, hyphen, and underscore. Can\'t start with a special char. Max length is 36 chars.')
     ->param('name', '', new Text(128), 'Provider name.')
-    ->param('from', '', new Phone(), 'Sender Phone number. Format this number with a leading \'+\' and a country code, e.g., +16175551212.', true)
-    ->param('senderId', '', new Text(0), 'Msg91 Sender ID.', true)
-    ->param('authKey', '', new Text(0), 'Msg91 Auth Key.', true)
+    ->param('templateId', '', new Text(0), 'Msg91 template ID', true)
+    ->param('senderId', '', new Text(0), 'Msg91 sender ID.', true)
+    ->param('authKey', '', new Text(0), 'Msg91 auth key.', true)
     ->param('enabled', null, new Boolean(), 'Set as enabled.', true)
     ->inject('queueForEvents')
     ->inject('dbForProject')
     ->inject('response')
-    ->action(function (string $providerId, string $name, string $from, string $senderId, string $authKey, ?bool $enabled, Event $queueForEvents, Database $dbForProject, Response $response) {
+    ->action(function (string $providerId, string $name, string $templateId, string $senderId, string $authKey, ?bool $enabled, Event $queueForEvents, Database $dbForProject, Response $response) {
         $providerId = $providerId == 'unique()' ? ID::unique() : $providerId;
 
         $options = [];
+        $credentials = [];
 
-        if (!empty($from)) {
-            $options['from'] = $from;
+        if (!empty($templateId)) {
+            $credentials['templateId'] = $templateId;
```

Comment:
Should this be in $options since it doesn't need to be encrypted?

Reply:
Yea makes sense. lets be consistent with what we use in other places

Reply:
@christyjacob4, oye, I'm not sure what the convention was when this was first created. For example, `isEuRegion` (a bool) is in `$credentials`:

https://github.com/appwrite/appwrite/blob/644d078f7f7054ffd0616c35d4df028374b5cd1b/app/controllers/api/messaging.php#L82-L84

Also host and port:

https://github.com/appwrite/appwrite/blob/644d078f7f7054ffd0616c35d4df028374b5cd1b/app/controllers/api/messaging.php#L246-L254

Let's just leave it.